src/mono/netcore/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.Mono.cs

```diff
@@ -1,16 +1,18 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
+using System.Runtime.CompilerServices;
+
 
 namespace System.Globalization
 {
     internal partial class GlobalizationMode
     {
-        internal static bool UseNls => true;
-
-        private static bool GetGlobalizationInvariantMode()
-        {
-            return GetInvariantSwitchValue();
+        internal static bool Invariant { get; } = GetInvariantSwitchValue();
+        internal static bool UseNls {
+            get {
+                return !Invariant && Interop.Globalization.LoadICU() == 0;
+            }
         }
```

Comment:
Since you're doing this in the `Getter` this will cause that every-time we check the `Nls` switch in windows, we call the native `LoadICU` API. We don't need that, we need to cache the value just like we do for `Invariant`.

```suggestion
        internal static bool UseNls { get; } = !Invariant && Interop.Globalization.LoadICU() == 0;
```

Reply:
Also, customers can disable using ICU and enable using NLS via a config switch. Just the same way as Invariant mode. So before calling `Interop.Globalization.LoadICU` we should check for the environment variable to see if it is set. `DOTNET_SYSTEM_GLOBALIZATION_USENLS`